www/src/templates/template-docs-markdown.js

```diff
@@ -16,6 +16,26 @@ import Breadcrumb from "../components/docs-breadcrumb"
 import Container from "../components/container"
 import PrevAndNext from "../components/prev-and-next"
 
+import { useActiveHash } from "../hooks/use-active-hash"
+
+const getHeadingIds = (toc, traverseFullDepth = false) => {
+  const idList = []
+  const hashToId = str => str.slice(1)
```

Comment:
Found an issue here on the `/contributing/docs-templates/` page. Getting:

```
TypeError: Cannot read property 'slice' of undefined
```

Noticed this while trying to build the site locally with `gatsby build`. That might be a good way to detect other pages causing problems and make sure builds aren't broken by these changes.

Reply:
maybe because `Why use templates?` is a child with a (not visible? parent), and this parent has an item can have not an `url`? 

the same case we have if there is a missing header level, then an empty element is attached  

see something similar i had with this PR:

- #19851 `fix(compontent): fix links with undefined in docs-table-of-contents.js`

example from this PR and there is no `url`:

```javascript
{ 
  url: '#2header', 
  title: '2. Header', 
  items:
    [
      { 
        items: 
	      [ 
		    { 
              url: '#4header', 
              title: '4. Header' 
            }
          ]
      }
    ]
}
```

Reply:
Apparently this is exactly the same problem. I am accounting for it in 1376382